utils/get-cstyle-files.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright 2022, Intel Corporation
+#
+
+"""
+Helper script for cstyle, which recursively obtains
+all of the files with .h or .c extensions, starting in
+the root directory passed as a parameter.
+Excluded files are containted in the .cstyleignore file.
+"""
+
+import os
+import sys
+import re
+
+if len(sys.argv) < 3:
+    print("Wrong number of arguments!"
+        "Usage: get-cstyle-files.py root_dir build_dir")
+    sys.exit(1)
+
+root_directory = sys.argv[1]
+build_directory = sys.argv[2]
+
+cstyle_ignore_f = os.path.join(root_directory, "utils", ".cstyleignore")
+
+with open(cstyle_ignore_f) as file:
+    exclude_file_paths = file.readlines()
+    exclude_file_paths = [exclude_file_path.rstrip() for 
+        exclude_file_path in exclude_file_paths]
+
+# replace '*' wildcard with '.+' for Python to understand
+for exclude_file_path in exclude_file_paths:
```

Comment:
Yes, I think it will be simpler to just stick to regex and not bother with wildcards. Especially since I've only translated '*' and I wouldn't want this script to get too complicated. I also had to change it a bit because I've noticed it was possible to just give a file name in .cstyleignore file (without using regular expressions or wildcards) for cstyle to ignore all of the files with this file name, which seemed a bit unclear and error prone so I enforced that the exact file path must be given (with usage of the regular expressions).